src/main.ts

```diff
@@ -74,7 +71,12 @@ async function run() {
     for (const pattern of patterns) {
       core.startGroup(`grep "${pattern.pattern}"`);
       const files = await getFiles(onlyChanged, changedFiles, pattern);
-      const results = await grep(pattern.syntax, pattern.pattern, files);
+      const results = await grep(
+        pattern.syntax,
+        pattern.binaryFiles,
+        pattern.pattern,
```

Comment:
This starting to look like a DataClump where we could add a `GrepOptions{syntax,binaryFiles,pattern}` object to pass a single argument of.

Not suggesting we do it now, just noting the smell.

Reply:
I did this slightly different to your suggestion to emphasize the pattern/files arguments against other arguments that have defaults